Add new chain to givEconomy guide (#106)
Giveth/giv-token-contracts: add-new-chain-to-givEconomy-guide.md

```
in below file https://github.com/Giveth/giveconomy-foundry/blob/develop/script/deployTokenDistro.s.sol

6. Fill `.env` file in the project with these data
```
PRIVATE_KEY=// I should start with 0x
ETHERSCAN_API_KEY= // from block explorer of that specific chain
VERIFIER_URL=https://api-cardona-zkevm.polygonscan.com/api // You can find it in block explorer
//  like https://docs.polygonscan.com/v/polygon-zkevm/getting-started/endpoint-urls

```
7. Edit https://github.com/Giveth/giveconomy-foundry/blob/develop/foundry.toml and add this line(for zkevm cardona,
for other chains you should change them based on the chain

```
cardona = { key = "${ETHERSCAN_API_KEY}", url = "https://api-cardona-zkevm.polygonscan.com/api", chain = 2442 }
```

8. Enter this command to check everything is working locally

`forge script script/deployTokenDistro.s.sol  --rpc-url  {rpcUrl}  --legacy --chain 2442 {Desired chainId}`

9. Enter this command to broadcast and verify the smart contracts on blockchain

`forge script script/deployTokenDistro.s.sol  --rpc-url  {rpcUrl}    --verify --broadcast --legacy --chain 2442 {Desired chainId}`

> 9. Verify contract with proxy admin in the block explorer

Reviewer: ```suggestion
9. Open the block-explorer website's smart contract page (code section) and declare it as a proxy.
<img width="278" alt="image" src="https://github.com/user-attachments/assets/a9dec69a-04ef-4807-b528-cfb3d9ce533b">

```

> 

Reviewer: Permission adding/revoking can be added as part of ownership transfer

2. Deploy it with foundry
`forge script script/deployUnipoolGIVpower.s.sol  --rpc-url  {rpcUrl}   --legacy --chain 2442 --verify --broadcast`

3. Verify contract with proxy admin in the block explorer

4. We call `notifyRewardAmount` method and set some reward for the contract https://cardona-zkevm.polygonscan.com/address/0x7e9f30a74fcdf035018bc007f9930aa171863e33#writeProxyContract the `rewardDistributor` address can call it (on unipool smart contract)

5. Call `transferOwnership` if it's needed

6. Call `setRewardDistribution` on Unipool GivPower to set the wallet address which will call the `notifyRewardAmount`

7. Check the address of `DISTRIBUTOR_ROLE` in tokenDistro contract https://cardona-zkevm.polygonscan.com/address/0x2df3e67be4e441cddd2d29c3d41dfd7d516f18e6#readProxyContract

8. Call `grantRole` in `tokenDistro` https://cardona-zkevm.polygonscan.com/address/0x2df3e67be4e441cddd2d29c3d41dfd7d516f18e6#writeProxyContract and grant `DISTRIBUTOR_ROLE` to givPower unipool address

9. Call `assign()` method in tokenDistro and assign the reward token amount to unipool givPower

## Test
1. Call `stake` method in unipoolGivPower contract

> 2. See the giviverse expansion percentage in https://giveth.io/givstream

Reviewer: giviverse?!

Author: <img width="961" alt="Screenshot 1403-07-16 at 1 12 16 in the afternoon" src="https://github.com/user-attachments/assets/23e7d63c-f659-47e4-8775-b593e30ce3d4">
Yes

3. Call the `getReward()` method in unipool givPower contract. the `result * percentage = the amount we show in harvest modal`

## Add subgraph configs
Should add new network in https://github.com/Giveth/giveconomy-subgraph/blob/develop/networks.yaml with needed addresses (all addresses should be in the notion file)

## Integrate with Frontend
You can know how to do it with looking at below PR
https://github.com/Giveth/giveth-dapps-v2/pull/4562/files
(LM adress is the UnipoolGivPower address, for staging ENV you should add configs to `development.tsx` and for produciton you need to add the configs to `production.tsx`

## Notify Reward Bot
1. We have a wallet address that is in charge for this purpose, we have th private key of that address in the server, so we should make sure call`setRewardDistribution` on Unipool GivPower to set the wallet address of this private key as `reward distributor`
2. We write a script in `giv-token-contract` to call it with crontab jobs https://github.com/Giveth/giv-token-contracts/commit/5188586beed3e0f73c7e0b9bbaf653134184a215

> 3. We should transfer some monet the reward distributor address for the gas fee on the new chain that we want ( if reward distributor doesnt have gas, the no`notifyRewardAmount` tx can't be executed

Reviewer: ```suggestion
3. Transfer some gas to the reward distributor address if it doesn't have enough to make `notifyRewardAmount` transactions
```

4. Setup the crontab on the server to execute this script periodically

## Giv unlcok bot
1. Copy a folder of https://github.com/Giveth/givpower-bot in the server, then change config values to the new chain and bring up the docker compose, then this app will unlcok the GIVs when it's the time

> 2. Make sure that the wallet address of the private key that we put in the config have enough gas fee in it

Reviewer: ```suggestion
2. Make sure that the wallet address of the private key that we put in the config has enough gas in it
```
